### ISSUE TITLE
learning rate schedule epoch updates at check_val_every_n_epoch

### ISSUE THREAD

**Issue by hazedine** [PyTorchLightning/pytorch-lightning#7538]
## 🐛 Bug

The learning rate scheduler does not update every epoch, but rather every `check_val_every_n_epoch`

## Please reproduce using the BoringModel

https://colab.research.google.com/drive/17Q0H6J9gH_Xuhommm-ZNTiOiJ_Z5rG_n?usp=sharing

### Expected behavior

The `check_val_every_n_epoch` has no influence on the learning rate scheduler updates.

### Environment

* CUDA:
	- GPU:
		- Tesla T4
	- available:         True
	- version:           10.1
* Packages:
	- numpy:             1.19.5
	- pyTorch_debug:     False
	- pyTorch_version:   1.8.1+cu101
	- pytorch-lightning: 1.3.1
	- tqdm:              4.41.1
* System:
	- OS:                Linux
	- architecture:
		- 64bit
		- 
	- processor:         x86_64
	- python:            3.7.10
	- version:           #1 SMP Tue Apr 20 19:55:43 PDT 2021

**Comment by rohitgr7** [comment]
duplicate: https://github.com/PyTorchLightning/pytorch-lightning/issues/6616
PR: https://github.com/PyTorchLightning/pytorch-lightning/pull/7032
